src/scheduled/index.js

```diff
@@ -0,0 +1,95 @@
+let { env, maybeHydrate, readArc } = require('../helpers')
+let hydrate = require('@architect/hydrate')
+let runner = require('./_runner')
+let series = require('run-series')
+let readline = require('readline')
+
+/**
+ * Looks for Scheduled Events and runs them as if they were cloudwatch events:
+ */
+module.exports = function createSchedule () {
+  let { arc } = readArc()
+
+  if (!arc.scheduled) {
+    return
+  }
+
+  let scheduled = {}
+  let schedulerBus
+
+  function keypress (input) {
+    if (input === 'T') {
+      schedulerBus && schedulerBus.runEvents()
+    }
+  }
+
+  scheduled.start = function start (options, callback) {
+    let { all, update } = options
+
+    series([
+      function _env (callback) {
+        if (!all) env(options, callback)
+        else callback()
+      },
+
+      // Loop through functions and see if any need dependency hydration
+      function _maybeHydrate (callback) {
+        if (!all) maybeHydrate(callback)
+        else callback()
+      },
+
+      // ... then hydrate Architect project files into functions
+      function _hydrateShared (callback) {
+        if (!all) {
+          hydrate({ install: false }, function next (err) {
+            if (err) callback(err)
+            else {
+              update.done('Project files hydrated into functions')
+              callback()
+            }
+          })
+        }
+        else callback()
+      },
+
+      function _finalSetup (callback) {
+        try {
+          schedulerBus = runner(arc.scheduled, update)
+        }
+        catch (e) {
+          callback(e)
+        }
+
+        readline.emitKeypressEvents(process.stdin)
```

Comment:
Not sure what the implications are from running this from inside the service module, as opposed to from inside the CLI module. Inside the CLI module we can more or less guarantee it's all human/user interaction; anywhere else makes me a bit nervous. wdyt?

Reply:
Yeah i was not very happy with this either
What do you think about something like this?
https://github.com/architect/sandbox/pull/466/commits/c7994a1c3a237850752bac1cc7c39b86f914cba2